job_bundles/Using_Files_In_Jobs.md

```diff
@@ -0,0 +1,775 @@
+
+for d in /shared/common /shared/projects/project1 /shared/projects/project2; do
+  echo "File contents for $d" > ${d}/file.txt
+done
+```
+
+Next, add an asset references file to the job bundle that includes all of these files as inputs for the job:
+
+```yaml
+cat > ${HOME}/job_attachments_devguide/asset_references.yaml << EOF
+assetReferences:
+  inputs:
+    filenames:
+    - /shared/common/file.txt
+    directories:
+    - /shared/projects/project1
+    - /shared/projects/project2
+EOF
+```
+
+Then, configure the Deadline Cloud CLI to submit jobs with the `WSAll` storage profile, and submit the job bundle:
+
+```bash
+deadline config set settings.storage_profile_id $WSALL_ID
+
+deadline bundle submit --farm-id $FARM_ID --queue-id $QUEUE1_ID job_attachments_devguide/
+```
+
+Notice that 3 files are uploaded to Amazon S3 when the job is submitted. Download the manifest objects for the job from Amazon S3
+to see which files were uploaded:
+
+```bash
+for manifest in $( \
+  aws deadline get-job --farm-id $FARM_ID --queue-id $QUEUE1_ID --job-id $JOB_ID \
+    --query 'attachments.manifests[].inputManifestPath' \
+    | jq -r '.[]'
+); do
+  aws s3 cp s3://$Q1_S3_BUCKET/DeadlineCloud/Manifests/$manifest manifest.json
+  echo "Manifest object: $manifest"
+  cat manifest.json
+done
+```
+
+There is a single manifest file with contents:
+
+```json
+{
+    "hashAlg": "xxh128",
+    "manifestVersion": "2023-03-03",
+    "paths": [
+        {
+            "hash": "87cb19095dd5d78fcaf56384ef0e6241",
+            "mtime": 1721147454416085,
+            "path": "home/cloudshell-user/job_attachments_devguide/script.sh",
+            "size": 39
+        },
+        {
+            "hash": "af5a605a3a4e86ce7be7ac5237b51b79",
+            "mtime": 1721163773582362,
+            "path": "shared/projects/project2/file.txt",
+            "size": 44
+        }
+    ],
+    "totalSize": 83
+}
+```
+
+Looking at the result from the GetJob API for this manifest, then you will see that the `rootPath` is "/" for this manifest:
+
+```bash
+aws deadline get-job --farm-id $FARM_ID --queue-id $QUEUE1_ID --job-id $JOB_ID --query 'attachments.manifests[*]'
+```
+
+The `rootPath` of a set of input files is always the longest common subpath of those files. If your job was submitted from, say,
+Windows instead and you had inputs on different drives then you would see a separate `rootPath` for each drive.
+The paths in a manifest are always relative to the `rootPath` of the manifest, so the input files that were uploaded are:
+
+1. `/home/cloudshell-user/job_attachments_devguide/script.sh`: The script file within the job bundle.
+2. `/shared/projects/project2/file.txt`: The file within a `SHARED` filesystem location in the `WSAll` storage profile that is **not** in
+   the list of required filesystem locations on queue `Q1`.
+
+Notice that the files contained in filesystem locations `FSComm` and `FS1` are not in that list. This is because those filesystem locations
```

Comment:
Might be nice to have the actual filesystem locations for FSComm and FS1 here. I had to scroll up to remember what they were